ouroboros-network-framework/src/Ouroboros/Network/Subscription/Dns.hs

```diff
@@ -115,143 +116,104 @@ dnsResolve tracer getSeed withResolverFn peerStatesVar beforeConnect (DnsSubscri
 
          Right rs -> do
              withResolverFn rs $ \resolver -> do
-                 ipv6Rsps <- newEmptyTMVarIO
-                 ipv4Rsps <- newEmptyTMVarIO
-                 gotIpv6Rsp <- newTVarIO False
-
                  -- Though the DNS lib does have its own timeouts, these do not work
                  -- on Windows reliably so as a workaround we add an extra layer
                  -- of timeout on the outside.
                  -- TODO: Fix upstream dns lib.
                  --       On windows the aid_ipv6 and aid_ipv4 threads are leaked incase
                  --       of an exception in the main thread.
                  res <- timeout 20 $ do
-                          aid_ipv6 <- async $ resolveAAAA resolver gotIpv6Rsp ipv6Rsps
-                          aid_ipv4 <- async $ resolveA resolver gotIpv6Rsp ipv4Rsps
+                          aid_ipv6 <- async $ resolveAAAA resolver
+                          aid_ipv4 <- async $ resolveA resolver aid_ipv6
```

Comment:
I tried this out, but the tests started failing, see https://hydra.iohk.io/build/5433421/nixlog/9

After thinking about this for a bit, it turns out we can't use `withAsync` here after all. The reason being that this function relies on the threads (or actually just one) still running when it exits, because the results are delivered via a `SubscriptionTarget`, which provides results step by step as requested. So `async` seems to be needed here after all.

Reply:
hmm, that's quite subtle, we will refactor this later.